frontend/src/components/protectedRoutes.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { useContext } from 'react'
+import { Outlet } from 'react-router-dom'
+import { UserContext } from './UserContext'
+
+export const ProtectedRoutes = (): any => {
+  const [user, loadingStatus] = useContext(UserContext)
+
+  if (user !== undefined && user !== null) {
+    return (<Outlet />)
+  }
+  if (loadingStatus === 'unloaded' || loadingStatus === 'loading') {
+    return (<h1>Loading user data...</h1>)
```

Comment:
will this second and third if statements ever get run? won't the user have a nil value always while loading?

Reply:
the 2nd and 3rd if statements worked when I tested them. the 2nd statement is the same as `user === null` while loading, so you're correct